src/platform/src/Bridge/Ollama/OllamaResultConverter.php

```diff
@@ -73,7 +74,18 @@ public function doConvertCompletion(array $data): ResultInterface
             return new ToolCallResult(...$toolCalls);
         }
 
-        return new TextResult($data['message']['content']);
+        $result = new TextResult($data['message']['content']);
+
+        if (\array_key_exists('prompt_cache_key', $options)) {
+            $metadata = $result->getMetadata();
+
+            $metadata->add('cached', true);
+            $metadata->add('prompt_cache_key', $options['prompt_cache_key']);
+            $metadata->add('cached_prompt_count', $data['prompt_eval_count']);
+            $metadata->add('cached_completion_count', $data['eval_count']);
```

Comment:
Wouldn't it make sense to group this data into a DTO, like there is `TokenUsage`, and then add that DTO to _metadata_ or perhaps even reuse the said DTO?

Reply:
Yes `CacheUsage` would be a good fit